src/classes/STG_InstallScript.cls

```diff
@@ -337,25 +338,62 @@ global without sharing class STG_InstallScript implements InstallHandler {
-                orgSettings.Prevent_Account_Deletion__c = true; 
-            }
-            if (orgSettings.Prevent_Plan_Requirement_Deletion__c == null || orgSettings.Prevent_Plan_Requirement_Deletion__c != true) {
-                orgSettings.Prevent_Plan_Requirement_Deletion__c = true; 
-            }
-            upsert orgSettings;
+            orgSettings = enablePreventForAccountsAndPlanRequirement();
         }
+        if (isNewInstallOrPushUpgradedTo196) {
+            UTIL_Debug.debug('***Upgrading to version 1.96 or new installs');
+            orgSettings = enablePreventPPlanDeletionHierarchyCustomSettings();
+        }
+        upsert orgSettings;
+    }
+
+    /*****************************************************************************************************************************************************************
+    * @description Enables Prevent Account Deletion and Plan Requirement Deletion by default on Hierarchy Custom Settings for new installs and when upgrading to 1.95
+    * @return void
+    */
+    global static Hierarchy_Settings__c enablePreventForAccountsAndPlanRequirement() {
+        Hierarchy_Settings__c orgSettings = UTIL_CustomSettingsFacade.getSettings();
+        if (orgSettings.Prevent_Account_Deletion__c == null || orgSettings.Prevent_Account_Deletion__c != true) {
+            orgSettings.Prevent_Account_Deletion__c = true;
+        }
+        if (orgSettings.Prevent_Plan_Requirement_Deletion__c == null || orgSettings.Prevent_Plan_Requirement_Deletion__c != true) {
+            orgSettings.Prevent_Plan_Requirement_Deletion__c = true;
+        }
+        return orgSettings;
+    }    
+
+    /*****************************************************************************************************************************************
+    * @description Enables Prevent Program Plan Deletion by default on Hierarchy Custom Settings for new installs and when upgrading to 1.96
+    * @return void
+    */
+    global static Hierarchy_Settings__c enablePreventPPlanDeletionHierarchyCustomSettings() {
+        Hierarchy_Settings__c orgSettings = UTIL_CustomSettingsFacade.getSettings();
+        if (orgSettings.Prevent_Program_Plan_Deletion__c == null || orgSettings.Prevent_Program_Plan_Deletion__c != true) {
+            orgSettings.Prevent_Program_Plan_Deletion__c = true;
+        }
+        return orgSettings;
+    }
+
+    /*****************************************************************************************************************************************
+    * @description Method to check if the package is being newly installed or push upgraded to a particular version
+    * @return true if it is either a new install or push upgraded to certain version else return false
+    */
+    @testVisible
+    private static boolean newInstallOrPushUpgrade(Version previousVersion, Version newVersion) {
```

Comment:
Consider renaming this to isNewInstallOrPushUpgrade or something similar to make it clear that this is more of a check than a function that performs changes.